File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int l = 0;
        int r = heights.size()-1;
        int m = INT_MIN;
        while(l<r){
            int area = (r-l) * min(heights[r],heights[l]);
            
            m = max(area,m);
            cout<<m<<endl;

            if(heights[l]<=heights[r]){
                l++;               
            }else{
                r--;
            }
        }

        return m;
        
    }
};

File: Data Structures & Algorithms/insert-new-interval/submission-2.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        # we can add new interval to list
        # sort by first element
        # merge them
        ans = []
        i = 0
        # cur end smaller than new start
        while i < len(intervals) and intervals[i][1] < newInterval[0]:
            ans.append(intervals[i])
            i += 1

        # cur start <= new end
        while i < len(intervals) and intervals[i][0] <= newInterval[1]:
            newInterval[0] = min(newInterval[0],intervals[i][0])
            newInterval[1] = max(newInterval[1],intervals[i][1])
            i += 1

        ans.append(newInterval)
        # remaining
        while i < len(intervals):
            ans.append(intervals[i])
            i+=1
        return ans
